File: backend/app/schemas/financial_analyzer.py
from datetime import date
from decimal import Decimal

from pydantic import BaseModel


class ExpenseCategorySummary(BaseModel):
    category: str
    amount: Decimal
    count: int


class FinancialAnalyzerResponse(BaseModel):
    start_date: date | None
    end_date: date | None

    # ========================================================
    # COUNTS
    # ========================================================

    invoice_count: int
    credit_note_count: int
    expense_count: int

    finished_product_count: int

    # ========================================================
    # SALES
    # ========================================================

    gross_sales: Decimal
    credit_notes: Decimal
    net_sales: Decimal

    # ========================================================
    # PRODUCTION COST
    # ========================================================

    actual_material_cost: Decimal
    actual_operation_cost: Decimal
    total_production_cost: Decimal

    # ========================================================
    # COMPANY EXPENSES
    # ========================================================

    total_expenses: Decimal

    # ========================================================
    # TOTAL BUSINESS COST / PROFIT
    # ========================================================

    total_business_cost: Decimal
    net_profit: Decimal

    # ========================================================
    # EXPENSE BREAKDOWN
    # ========================================================

    expense_breakdown: list[
        ExpenseCategorySummary
    ]

File: frontend/src/types/financialAnalyzer.ts
export interface ExpenseCategorySummary {
  category: string;
  amount: string;
  count: number;
}


export interface FinancialAnalyzerResponse {
  start_date: string | null;
  end_date: string | null;

  // ========================================================
  // COUNTS
  // ========================================================

  invoice_count: number;
  credit_note_count: number;
  expense_count: number;

  finished_product_count: number;

  // ========================================================
  // SALES REVENUE
  // ========================================================

  gross_sales: string;
  credit_notes: string;
  net_sales: string;

  // ========================================================
  // PRODUCTION COST
  // ========================================================

  actual_material_cost: string;
  actual_operation_cost: string;
  total_production_cost: string;

  // ========================================================
  // COMPANY EXPENSES
  // ========================================================

  total_expenses: string;

  // ========================================================
  // TOTAL BUSINESS COST / PROFIT
  // ========================================================

  total_business_cost: string;
  net_profit: string;

  // ========================================================
  // BREAKDOWN
  // ========================================================

  expense_breakdown: ExpenseCategorySummary[];
}